### PR TITLE
feat: add STS Accounts API

### PR DESCRIPTION
**Pull request by paullatzelsperger**
## What this PR changes/adds

adds the STS Accounts API (+controller) to manage STS Accounts (formerly known as StsClients). To achieve that, the `StsAccountService` was extended to accommodate the new functionality like updating, deleting etc.

## Why it does that

a way to manage STS Accounts is required

## Further notes

- a significant portion of the changeset was caused by the renaming `StsClient...` -> `StsAccount...`
- the STS Accounts API comes with Token-based Auth by default, but that can be customized by registering an `AuthenticationService` for the `"sts-accounts-api"` context

## Linked Issue(s)

Closes # <-- _insert Issue number if one exists_

_Please be sure to take a look at the [contributing guidelines](https://github.com/eclipse-edc/.github/blob/main/CONTRIBUTING.md#submit-a-pull-request) and our [etiquette for pull requests](https://github.com/eclipse-edc/.github/blob/main/contributing/pr_etiquette.md)._

### CODE REVIEW

File: extensions/common/iam/identity-trust/identity-trust-sts/identity-trust-sts-accounts-api/src/main/java/org/eclipse/edc/api/iam/identitytrust/sts/accounts/controller/StsAccountsApiController.java [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+import org.eclipse.edc.iam.identitytrust.sts.spi.service.StsAccountService;
+import org.eclipse.edc.spi.query.QuerySpec;
+
+import java.util.Collection;
+
+import static org.eclipse.edc.web.spi.exception.ServiceResultHandler.exceptionMapper;
+
+@Consumes(MediaType.APPLICATION_JSON)
+@Produces(MediaType.APPLICATION_JSON)
+@Path("/v1/accounts")
```
Comment (ndr-brt) [inline extensions/common/iam/identity-trust/identity-trust-sts/identity-trust-sts-accounts-api/src/main/java/org/eclipse/edc/api/iam/identitytrust/sts/accounts/controller/StsAccountsApiController.java:39]: shouldn't it be v1alpha?

File: extensions/common/iam/identity-trust/identity-trust-sts/identity-trust-sts-accounts-api/src/main/resources/sts-accounts-api-version.json [hunk 2]
```diff
@@ -0,0 +1,7 @@
+[
+  {
+    "version": "1.0.0",
```
Comment (ndr-brt) [inline extensions/common/iam/identity-trust/identity-trust-sts/identity-trust-sts-accounts-api/src/main/resources/sts-accounts-api-version.json:3]:
```suggestion
    "version": "1.0.0-alpha",
```

File: extensions/common/iam/identity-trust/identity-trust-sts/identity-trust-sts-accounts-api/src/main/resources/sts-accounts-api-version.json [hunk 3]
```diff
@@ -0,0 +1,7 @@
...
+    "urlPath": "/v1alpha",
+    "lastUpdated": "2024-09-24T14:14:00Z"
```
Comment (ndr-brt) [inline extensions/common/iam/identity-trust/identity-trust-sts/identity-trust-sts-accounts-api/src/main/resources/sts-accounts-api-version.json:5]: `maturity` is missing

Comment (paullatzelsperger) [reply extensions/common/iam/identity-trust/identity-trust-sts/identity-trust-sts-accounts-api/src/main/resources/sts-accounts-api-version.json:5]: i thought we can't set `maturity`, unless it is `stable`?

File: extensions/common/iam/identity-trust/identity-trust-sts/identity-trust-sts-accounts-api/src/main/java/org/eclipse/edc/api/iam/identitytrust/sts/accounts/controller/StsAccountsApiController.java (hunk 1 above)
Comment (paullatzelsperger) [reply extensions/common/iam/identity-trust/identity-trust-sts/identity-trust-sts-accounts-api/src/main/java/org/eclipse/edc/api/iam/identitytrust/sts/accounts/controller/StsAccountsApiController.java:39]: it definitely should, thanks

File: extensions/common/iam/identity-trust/identity-trust-sts/identity-trust-sts-accounts-api/src/main/resources/META-INF/services/org.eclipse.edc.spi.system.ServiceExtension [hunk 4]
```diff
@@ -0,0 +1,16 @@
+#
+#  Copyright (c) 2020 - 2022 Bayerische Motoren Werke Aktiengesellschaft (BMW AG)
```
Comment (wolf4ood) [inline extensions/common/iam/identity-trust/identity-trust-sts/identity-trust-sts-accounts-api/src/main/resources/META-INF/services/org.eclipse.edc.spi.system.ServiceExtension:2]:
```suggestion
#  Copyright (c) 2024 Bayerische Motoren Werke Aktiengesellschaft (BMW AG)
```

File: extensions/common/store/sql/sts-client-store-sql/src/main/java/org/eclipse/edc/iam/identitytrust/sts/store/SqlStsClientStoreExtension.java [hunk 5]
```diff
@@ -32,7 +32,7 @@
 
 import static org.eclipse.edc.transaction.datasource.spi.DataSourceRegistry.DEFAULT_DATASOURCE;
 
-@Provides({ StsClientStore.class })
+@Provides({ StsAccountStore.class })
 @Extension(value = "SQL sts client store")
```
Comment (wolf4ood) [inline extensions/common/store/sql/sts-client-store-sql/src/main/java/org/eclipse/edc/iam/identitytrust/sts/store/SqlStsClientStoreExtension.java:36]:
```suggestion
@Extension(value = "SQL sts accounts store")
```

File: spi/common/identity-trust-sts-spi/src/main/java/org/eclipse/edc/iam/identitytrust/sts/spi/service/StsTokenGenerationProvider.java [hunk 6]
```diff
@@ -27,10 +27,10 @@
...
     /**
-     * Returns a {@link TokenGenerationService} for the input {@link StsClient}
+     * Returns a {@link TokenGenerationService} for the input {@link StsAccount}
      *
-     * @param client The {@link StsClient}
+     * @param client The {@link StsAccount}
      * @return The {@link TokenGenerationService}
      */
-    TokenGenerationService tokenGeneratorFor(StsClient client);
+    TokenGenerationService tokenGeneratorFor(StsAccount client);
```
Comment (wolf4ood) [inline spi/common/identity-trust-sts-spi/src/main/java/org/eclipse/edc/iam/identitytrust/sts/spi/service/StsTokenGenerationProvider.java:35]:
```suggestion
    TokenGenerationService tokenGeneratorFor(StsAccount stsAccount);
```

**Review by wolf4ood — APPROVED** [review]
just minor nits

File: extensions/common/iam/identity-trust/identity-trust-sts/identity-trust-sts-accounts-api/src/main/resources/sts-accounts-api-version.json (hunk 3 above)
Comment (ndr-brt) [reply extensions/common/iam/identity-trust/identity-trust-sts/identity-trust-sts-accounts-api/src/main/resources/sts-accounts-api-version.json:5]: maybe you're right... I don't remember :smiling_face_with_tear: 

EDIT: you're right

**Review by ndr-brt — APPROVED** [review]

**Pull request merged**